code/game/machinery/kitchen/microwave.dm

```diff
@@ -410,6 +410,11 @@
 	return ffuu
 
 /obj/machinery/microwave/Topic(href, href_list)
+	if(stat & BROKEN)
+		return
+	if(href_list["action"] == "dispose")

```

Comment:
This should be in the switch at line 421.

And all of these should be after the `if (..())` check. Which is usually a usability and security call.

Reply:
Thats is the problem.
the ..() can not be before the action check. As it also checks for no power.

`/obj/machinery/CanUseTopic(var/mob/user)
	if(stat & BROKEN)
		return STATUS_CLOSE

	if(!interact_offline && (stat & NOPOWER))
		return STATUS_CLOSE

	return ..()`

Reply:
Yeah, me and Arrow couldn't come up with code that will have it in a switch. as `..()` will just quit proc before switch if no power